cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup_test.go

```diff
@@ -472,20 +473,22 @@ func TestNodeGroupDecreaseTargetSize(t *testing.T) {
 
 	t.Run("MachineSet", func(t *testing.T) {
 		tc := testCase{
-			description: "decrease by 1",
-			initial:     3,
-			expected:    2,
-			delta:       -1,
+			description:   "nodes targetSize:3 delta:-1 existingNodes: 3. Error when trying to delete an existing node",
```

Comment:
Changing the semantics of this test means we no longer have any code coverage for this line:

https://github.com/openshift/kubernetes-autoscaler/blob/master/cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup.go#L180

Can we keep/rearrange the test input so that we still have coverage here?

Reply:
In fact, we effectively now no longer test the real semantics of this function - the only part of the function that decreases the target size is the last line and that is now never called (in the unit tests).

Reply:
Updated. DecreaseTargetSize should only decrease the size when the current target size of the nodeGroup is bigger than the number existing instances for that group. We force such a scenario with targetSizeIncrement.